src/Database/LSMTree/Internal/Normal.hs

```diff
@@ -7,10 +7,10 @@ module Database.LSMTree.Internal.Normal (
 import           Control.DeepSeq (NFData (..))
 
 -- | Result of a single point lookup.
-data LookupResult k v blobref =
-    NotFound      !k
-  | Found         !k !v
-  | FoundWithBlob !k !v !blobref
+data LookupResult v blobref =
+    NotFound
+  | Found         !v
+  | FoundWithBlob !v !blobref
```

Comment:
I think that makes things a bit nicer and in many places we can get away without having to allocate tuples `(k, res)` instead.